code/datums/components/simple_access.dm

```diff
@@ -0,0 +1,32 @@
+///This component allows us to give a mob access without giving them an ID card.
+/datum/component/simple_access
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	///list of accesses we are allowed to access via this component
+	var/list/access
+
+/datum/component/simple_access/Initialize(list/new_access, atom/donor_atom)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	access = new_access
+	RegisterSignal(parent, COMSIG_MOB_TRIED_ACCESS, .proc/on_tried_access)
+	if(!donor_atom)
+		return
+	if(istype(donor_atom, /obj/item/organ))
+		RegisterSignal(donor_atom, COMSIG_ORGAN_REMOVED, .proc/on_donor_removed)
```

Comment:
should also register to donor being deleted

Reply:
Good catch